reward/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+            }
+        })
+        .for_each(drop);
+
+    term::headline("Retrieving commits...");
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Commit"),
+        term::format::bold("Signature"),
+        term::format::bold("ENS"),
+    ]);
+
+    let mut head_sha1 = None;
+    // show commits in provided range and collect all ens which have contributed
+    let all_ens = revwalk
+        .filter_map(|sha1| {
+            let sha1 = match sha1 {
+                Err(_) => return None,
+                Ok(sha1) => sha1,
+            };
+            if head_sha1.is_none() {
+                head_sha1 = Some(sha1);
+            }
+
+            let ens = {
+                let fp = git::commit_ssh_fingerprint(Path::new("."), &sha1.to_string()).ok()?;
+                if fp.is_none() {
+                    table.push([
+                        term::format::highlight(sha1),
+                        term::format::tertiary("missing"),
+                        term::format::secondary("missing"),
+                    ]);
+                    return None;
+                }
+                let fp = fp?;
+                let ens = ssh_to_ens.get(&fp);
+                table.push([
+                    term::format::highlight(sha1),
+                    term::format::tertiary(fp.clone()),
+                    term::format::secondary(
+                        ens.map(|ens| ens.name.to_owned())
+                            .unwrap_or_else(|| String::from("missing")),
+                    ),
+                ]);
+                ens
+            };
+
+            ens
+        })
+        .collect::<Vec<_>>();
+    table.render();
+    term::blank();
+
+    if all_ens.is_empty() {
+        return Err(anyhow!(
+            "No contributor with a set ENS was found in the given range"
+        ));
+    }
+
+    let currency = options.currency.unwrap_or_else(|| {
+        term::select_with_prompt(
+            "What's the reward's currency?",
+            &["wETH", "DAI", "USDC", "USDT", "Other"],
+            &"wETH",
+        )
+        .unwrap()
+        .to_string()
+    });
+
+    let (mut decimals, mut address) = match currency.as_str() {
+        "wETH" => (18, "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2".to_string()),
+        "DAI" => (18, "0x6b175474e89094c44da98b954eedeac495271d0f".to_string()),
+        "USDC" => (6, "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48".to_string()),
+        "USDT" => (6, "0xdac17f958d2ee523a2206206994597c13d831ec7".to_string()),
+        "Other" => (18, "0x0000000000000000000000000000000000000000".to_string()),
+        _ => {
+            // Supplied token e.g. `--currency 0x2b591e99afe9f32eaa6214f7b7629768c40eeb39`
+            let token: ERC20<SignerMiddleware<_, _>> = ERC20::new(signer.clone(), &currency);
+            let decimals = rt.block_on(token.get_decimals())?.into();
+
+            (decimals, currency.to_string())
+        }
+    };
```

Comment:
This whole thing would be cleaner with an `enum Token` that has methods for returning the addresses/decimals/symbols